linux/src/threadpool_linux.c

```diff
@@ -291,6 +308,46 @@ static int reallocate_threadpool_array(THREADPOOL* threadpool)
     return result;
 }
 
+static TIMER_INSTANCE* get_next_timer_instance(THREADPOOL* threadpool)
+{
+    TIMER_INSTANCE* result;
+
+    // Make sure only one thread access the loop at once
+    do
+    {
+        int32_t current_val = interlocked_compare_exchange(&threadpool->timer_instance_access, 1, 0);
+        if (current_val == 0)
+        {
+            break;
+        }
+        else
+        {
+            // Do Nothing wait for address
+        }
+        (void)wait_on_address(&threadpool->timer_instance_access, current_val, UINT32_MAX);
+    } while (true);
```

Comment:
Actually we should be able to do this without a lock.

```
for (...)
{
    if (interlocked_compare_exchange(&threadpool->timer_instance[index].timer_status, TIMER_ENABLED, TIMER_DISABLED) == TIMER_DISABLED)
...
}
```

First one to switch a state from disabled to enabled will get exclusive access of that slot.

Simplifies this function quite a bit.

Reply:
Good point, I'll change this.